store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java

```diff
@@ -335,10 +335,12 @@ private CarbonTable buildCarbonTable() {
       // When dimensions are default set to sort column,
       // Inverted index will be supported by default for sort columns.
       for (Field field : schema.getFields()) {
-        if (field.getDataType() == DataTypes.STRING ||
-            field.getDataType() == DataTypes.DATE ||
-            field.getDataType() == DataTypes.TIMESTAMP) {
-          sortColumnsList.add(field.getFieldName());
+        if (null != field) {
```

Comment:
Please add comment when field can be null

Reply:
Fixed. added comment